src/crimson/osd/ops_executer.h

```diff
@@ -178,10 +178,21 @@ auto OpsExecuter::with_effect(
 
 template <typename Func>
 OpsExecuter::osd_op_errorator::future<> OpsExecuter::submit_changes(Func&& f) && {
+  if (obc/*must be an osd op*/) {
```

Comment:
At the moment `submit_changes()` is not called in conjunction with `execute_pg_op()`, so this can be an assert. In long term we need to clean-up the PG-vs-OSD-ops.